src/main/groovy/com/github/benmanes/gradle/versions/updates/resolutionstrategy/ComponentFilter.groovy

```diff
@@ -0,0 +1,7 @@
+package com.github.benmanes.gradle.versions.updates.resolutionstrategy
+
+interface ComponentFilter {
```

Comment:
Here you could add `@HasImplicitReceiver` like in `org.gradle.api.Action`, and could simplify the usage a little.

Reply:
Oh perfect, that's exactly what I wanted to do but didn't know. 
I think I am now done @ghus-raba